nuxeo-runtime/nuxeo-connect-standalone/src/main/java/org/nuxeo/connect/update/task/update/UpdateManager.java

```diff
@@ -369,20 +380,34 @@ public String getServerRelativePath(File someFile) {
      * Create a new entry in the registry given the entry key. A base version will be automatically created if needed.
      *
      * @param key
-     * @throws Exception
+     * @throws PackageException
      */
     public Entry createEntry(String key) throws PackageException {
         Entry entry = new Entry(key);
-        Match<File> m = JarUtils.findJar(serverRoot, key);
+        updateBaseVersion(entry);
+        registry.put(key, entry);
+        return entry;
+    }
+
+    /**
+     * Create a base version for the given entry if needed.
+     *
+     * @param entry
+     * @return true if a base version was actually created, false otherwise
+     * @throws PackageException
+     * @since 1.4.26
+     */
+    public boolean updateBaseVersion(Entry entry) throws PackageException {
```

Comment:
I wouldn't say "update": it is never "updated" in the meaning that the value is changed. Rather `createBaseVersion()` or `addBaseVersion()`?